File: flutter_app/lib/models/job.dart
import 'package:json_annotation/json_annotation.dart';

part 'job.g.dart';

@JsonSerializable()
class Job {
  final int id;
  final String title;
  final String salary;
  final String? city;
  final String? experience;
  final String? education;
  final String? jobType;
  final String? description;
  final List<String>? skills;
  final List<String>? welfare;
  final bool active;
  final bool hot;
  final int viewCount;
  final Company? company;

  Job({
    required this.id,
    required this.title,
    required this.salary,
    this.city,
    this.experience,
    this.education,
    this.jobType,
    this.description,
    this.skills,
    this.welfare,
    this.active = true,
    this.hot = false,
    this.viewCount = 0,
    this.company,
  });

  factory Job.fromJson(Map<String, dynamic> json) => _$JobFromJson(json);
  Map<String, dynamic> toJson() => _$JobToJson(this);
}

@JsonSerializable()
class Company {
  final int id;
  final String name;
  final String? logoUrl;
  final String? industry;
  final String? scale;
  final String? financing;
  final String? intro;
  final String? address;

  Company({
    required this.id,
    required this.name,
    this.logoUrl,
    this.industry,
    this.scale,
    this.financing,
    this.intro,
    this.address,
  });

  factory Company.fromJson(Map<String, dynamic> json) => _$CompanyFromJson(json);
  Map<String, dynamic> toJson() => _$CompanyToJson(this);
}

File: java_backend/src/main/java/com/boss/recruitment/entity/Job.java
package com.boss.recruitment.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "jobs")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Job {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String title;

    @Column(nullable = false, length = 50)
    private String salary;

    @Column(length = 50)
    private String city;

    @Column(length = 20)
    private String experience;

    @Column(length = 20)
    private String education;

    @Column(length = 20)
    private String jobType; // 全职、兼职、实习

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(length = 500)
    private String skills; // JSON格式存储技能标签

    @Column(length = 500)
    private String welfare; // JSON格式存储福利标签

    @Column(nullable = false)
    private Boolean active = true;

    @Column(nullable = false)
    private Boolean hot = false;

    private Integer viewCount = 0;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id")
    private Company company;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hr_id")
    private User hr;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
}
